salt/cloud/__init__.py

```diff
@@ -185,6 +185,10 @@ def __init__(self, path=None, opts=None, config_dir=None, pillars=None):
         else:
             self.opts = salt.config.cloud_config(path)
 
+        # Check the cache-dir exists. If not, create it.
+        if not os.path.exists(self.opts['cachedir']):
```

Comment:
Basically, I'm using the salt cloud API from python, without using the CLI or any intervening components, as part of a orchestration system.

Specifically, I'm basically treating it as a tool to deploy VMs, and run commands on them. 

I don't see why I should have to bother with salt's own custom config thing, when I'm perfectly happy (and a lot more familiar) just writing plain old python. 

see https://github.com/fake-name/AutoTriever/blob/master/marshaller/salt_runner.py and https://github.com/fake-name/AutoTriever/blob/master/marshaller/salt_scheduler.py